test/Sema/value_generics.swift

```diff
@@ -119,3 +119,65 @@ func testC4<let T: Int>(with c: C<T, T>) {
 }
 
 struct D<let N: Int & P> {} // expected-error {{non-protocol, non-class type 'Int' cannot be used within a protocol-constrained type}}
+
+struct E<A, let b: Int> { // expected-note {{'b' previously declared here}}
+  static var b: Int { // expected-error {{invalid redeclaration of 'b'}}
+                      // expected-note@-1 {{'b' declared here}}
+    123
+  }
+
+  let b: String // expected-note {{'b' previously declared here}}
+                // expected-note@-1 {{'b' declared here}}
+
+  func b() {} // expected-error {{invalid redeclaration of 'b()'}}
+              // expected-note@-1 {{'b' declared here}}
+
+  func dumb() -> Int {
+    Self.b // OK
+  }
+
+  static func dumb2() -> Int {
+    Self.b // OK
+  }
+}
+
+func testE1() -> Int {
+  E<Int, 123>.b // OK
```

Comment:
Does this/should this work in type context? eg, `E<Int, E<Int, 123>.b>`

Reply:
It does work now, but it seems the LSG did not intend for this and wants to disallow it for the time being. I will go ahead and make name lookup not return these generic parameters in type contexts.